src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/BaseComponent.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    /// <summary>
+    /// This is a base class for components
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public abstract class BaseComponent
+    {
+        internal IntPtr Handle;
+        private const string LogTag = "Tizen.Applications.BaseComponent";
+
+        /// <summary>
+        /// Component types.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public enum ComponentType
+        {
+            /// <summary>
+            /// The frame component.
+            /// </summary>
+            Frame = Interop.CBApplication.NativeComponentType.Frame,
+
+            /// <summary>
+            /// The service component.
+            /// </summary>
+            Service = Interop.CBApplication.NativeComponentType.Service
+        }
+
+        /// <summary>
+        /// Occurs when the system memory is low.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowMemoryEventArgs> LowMemory;
+
+        /// <summary>
+        /// Occurs when the system battery is low.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowBatteryEventArgs> LowBattery;
+
+        /// <summary>
+        /// Occurs when the system language is chagned.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LocaleChangedEventArgs> LocaleChanged;
+
+        /// <summary>
+        /// Occurs when the region format is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<RegionFormatChangedEventArgs> RegionFormatChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<DeviceOrientationEventArgs> DeviceOrientationChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<SuspendedStateEventArgs> SuspendedStateChanged;
+
+        /// <summary>
+        /// A component instance ID.
+        /// It will be created after OnCreate method is invoked.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string Id { get; internal set; }
```

Comment:
Who is set Id ?

Reply:
BaseFactory makes a BaseComponent Instance and assigns an ID.